ldk/javascript/src/utils/mapper.ts

```diff
@@ -0,0 +1,94 @@
+import { TextEncoder, TextDecoder } from 'text-encoding-shim';
+import { HTTPResponse, Socket } from '../network';
+
+export const mapToUint8Array = (data: ArrayBuffer): Uint8Array => new Uint8Array(data);
+
+export const mapToHttpResponse = (response: OliveHelps.HTTPResponse): HTTPResponse => ({
+  statusCode: response.statusCode,
+  body: new Uint8Array(response.body),
+  headers: response.headers,
+});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const handleCaughtError = (reject: (reason?: any) => void, error: Error, type: string): void => {
+  console.error(`Received error calling ${type}: ${error.message}`);
+  reject(error);
+};
+
+enum MessageType {
+  text = 1,
+  binary = 2,
+}
+
+const mapToBinaryData = (message: string | Uint8Array): Array<number> =>
+  typeof message === 'string' ? [...new TextEncoder().encode(message)] : [...message];
+
+const mapToResponseMessage = (messageType: MessageType, buffer: ArrayBuffer): string | Uint8Array => {
+  const data = mapToUint8Array(buffer);
+  return messageType === MessageType.text ? new TextDecoder().decode(data) : data;
+};
+
+const mapToMessageType = (message: string | Uint8Array): MessageType =>
+  typeof message === 'string' ? MessageType.text : MessageType.binary;
+
+export const mapToSocket = (socket: OliveHelps.Socket): Socket => ({
```

Comment:
These are all pretty straight forward functions except for this one. I think we should add some documentation about what this function is doing.